src/halmos/sevm.py

```diff
@@ -330,7 +327,7 @@ def pop(self) -> Word:
         return self.stack.pop()
 
     def dup(self, n: int) -> None:
-        self.push(self.stack[-n])
+        self.stack.append(self.stack[-n])
```

Comment:
this bypasses the `simplify` call in `self.push`, saves 5% on the maze benchmark